### ISSUE TITLE
Allow passing unbound methods to matchers when method isn't called

### ISSUE THREAD

**Issue by adalinesimonian** [jest-community/eslint-plugin-jest#985]
`jest/unbound-method` currently returns a problem for:

```js
import getMethod from './get-method'

const obj = {
  method: jest.fn(),
}

it('should get a method', () => {
  expect(getMethod(obj)).toBe(obj.method) // problem here: Avoid referencing unbound methods…
})
```

I think it'd make sense to allow passing methods to matchers to check for references in cases where the method isn't being called.

**Comment by G-Rath** [comment]
The problem is there's no way for us to know if a general matcher will call that argument or not - we could ship a list of all Jest matchers to at least special case them, but I'm a bit reluctant to do that and it would make us less consistent with the original rule (since that has the same issue).

We could do it for just a few of the matchers (like the equality ones) but I feel like this isn't a common enough pattern to be worth  the inconsistency.
